### PR TITLE
OCPVE-706: Secure metrics endpoints

### PR DESCRIPTION
**Pull request by suleymanakbas91**
This PR secures the metrics endpoints exposed in lvm-operator, vg-manager, and topolvm-node. For lvm-operator and vg-manager, this uses the new secure metrics serving feature introduced in https://github.com/kubernetes-sigs/controller-runtime/pull/2407. For topolvm-node, this PR introduces a sidecar container `kube-rbac-proxy` which handles the TLS connection.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -350,27 +350,13 @@ $ make undeploy
 
 ## Metrics
 
-The LVM Operator runs a metrics exporter sidecar to export Prometheus metrics. To enable monitoring on OpenShift clusters, assign the `openshift.io/cluster-monitoring` label to the same namespace that you deployed LVMS to.
+To enable monitoring on OpenShift clusters, assign the `openshift.io/cluster-monitoring` label to the same namespace that you deployed LVMS to.
```
Comment (jakobmoellerdev) [inline README.md:353]: what happens when the cluster monitoring is disabled? will we still have the certificates setup correctly, but just the scraping disabled?

File: cmd/operator/operator.go [hunk 2]
```diff
@@ -77,7 +78,7 @@ func NewCmd(opts *Options) *cobra.Command {
 	}
 
 	cmd.Flags().StringVar(
-		&opts.metricsAddr, "metrics-bind-address", DefaultMetricsAddr, "The address the metric endpoint binds to.",
+		&opts.diagnosticsAddr, "diagnostics-address", DefaultDiagnosticsAddr, "The address the diagnostics endpoint binds to.",
 	)
 	cmd.Flags().StringVar(
```
Comment (jakobmoellerdev) [inline cmd/operator/operator.go:83]: why schange this to diagnosticsadress? for me this is not clearer than metrics-bind-addresssince its literally setting up the metrics server

File: cmd/vgmanager/vgmanager.go [hunk 3]
```diff
@@ -67,7 +68,7 @@ func NewCmd(opts *Options) *cobra.Command {
 	}
 
 	cmd.Flags().StringVar(
-		&opts.metricsAddr, "metrics-bind-address", DefaultMetricsAddr, "The address the metric endpoint binds to.",
+		&opts.diagnosticsAddr, "diagnosticsAddr", DefaultDiagnosticsAddr, "The address the diagnostics endpoint binds to.",
```
Comment (jakobmoellerdev) [inline cmd/vgmanager/vgmanager.go:71]: see above regarding diagnostics instead of metrics

**Review by jakobmoellerdev — COMMENTED** [review]
Just one minor nit and some questions, overall looks great, this is a huge improvement! :tada:

File: README.md (hunk 1 above)
Comment (suleymanakbas91) [reply README.md:353]: The certificate setup is working independently of enabling monitoring, so everything works but just the scraping gets disabled.

File: cmd/operator/operator.go (hunk 2 above)
Comment (suleymanakbas91) [reply cmd/operator/operator.go:83]: To align naming with https://github.com/kubernetes-sigs/cluster-api/pull/9264.

**Review by jakobmoellerdev — APPROVED** [review]
/lgtm

**Pull request merged**